python/nav/web/auth/utils.py

```diff
@@ -43,6 +43,14 @@ def set_account(request, account, cycle_session_id=True):
     request.session.save()
 
 
+def clear_session(request):
+    """Clears the session and logs out the current account"""
+    if hasattr(request, "account"):
+        del request.account
+    request.session.flush()
+    request.session.save()
```

Comment:
Django itself doesn't bother with save after a flush.

Reply:
Yes, it seems SessionMiddleware handles saving the session. Removed save call

Reply:
This is true, Django lets SessionMiddleware handle saving during response processing. Some tests using the `client` fixture needs the session to be saved to work, and SessionMiddleware doesnt seem to be called during this automatically, so some changes to the tests will have to do be done. I'll open an issue for this and work on that separately, since it affects #2813 as well as this